server/server_test.go

```diff
@@ -0,0 +1,339 @@
+			Entry("sz", "1:1:w", "1:1:w"),
+		)
+
+		It("should fail with inavailable seccomp profile", func() {
+			// Given
+			gomock.InOrder(
+				serverMock.EXPECT().GetData().Times(2).Return(serverConfig),
+				serverMock.EXPECT().GetLibConfigIface().Return(libMock),
+				libMock.EXPECT().GetStore().Return(storeMock, nil),
+				libMock.EXPECT().GetData().Return(libConfig),
+			)
+			serverConfig.SeccompProfile = invalidDir
+
+			// When
+			server, err := server.New(context.Background(), serverMock)
+
+			// Then
+			Expect(err).NotTo(BeNil())
+			Expect(server).To(BeNil())
+		})
+
+		It("should fail with wrong seccomp profile", func() {
+			// Given
+			gomock.InOrder(
+				serverMock.EXPECT().GetData().Times(2).Return(serverConfig),
+				serverMock.EXPECT().GetLibConfigIface().Return(libMock),
+				libMock.EXPECT().GetStore().Return(storeMock, nil),
+				libMock.EXPECT().GetData().Return(libConfig),
+			)
+			serverConfig.SeccompProfile = "/dev/null"
+
+			// When
+			server, err := server.New(context.Background(), serverMock)
+
+			// Then
+			Expect(err).NotTo(BeNil())
+			Expect(server).To(BeNil())
+		})
+
+		It("should fail with invalid stream address and port", func() {
+			// Given
+			mockNewServer()
+			serverConfig.StreamAddress = invalid
+			serverConfig.StreamPort = invalid
+
+			// When
+			server, err := server.New(context.Background(), serverMock)
+
+			// Then
+			Expect(err).NotTo(BeNil())
+			Expect(server).To(BeNil())
+		})
+
+		It("should fail with invalid TLS certificates", func() {
+			// Given
+			mockNewServer()
+			serverConfig.StreamEnableTLS = true
+			serverConfig.StreamTLSCert = invalid
+			serverConfig.StreamTLSKey = invalid
+
+			// When
+			server, err := server.New(context.Background(), serverMock)
+
+			// Then
+			Expect(err).NotTo(BeNil())
+			Expect(server).To(BeNil())
+		})
+	})
+
+	t.Describe("CreateMetricsEndpoint", func() {
+		// Prepare the sut
+		BeforeEach(setupSUT)
+
+		It("should succeed", func() {
+			// Given
+
+			// When
+			mux, err := sut.CreateMetricsEndpoint()
+
+			// Then
+			Expect(err).To(BeNil())
+			Expect(mux).NotTo(BeNil())
+			sut.StopMonitors()
+		})
+	})
+
+	t.Describe("StartExitMonitor", func() {
+		// Prepare the sut
+		BeforeEach(setupSUT)
+
+		It("should succeed", func() {
+			// Given
+			go sut.StartExitMonitor()
+			closeChan := sut.MonitorsCloseChan()
+			Expect(closeChan).NotTo(BeNil())
+
+			// When
+			closeChan <- struct{}{}
+
+			// Then
+			Expect(closeChan).NotTo(BeNil())
+		})
+	})
+
+	t.Describe("Shutdown", func() {
+		// Prepare the sut
+		BeforeEach(setupSUT)
+
+		It("should succeed", func() {
+			// Given
+			gomock.InOrder(
+				storeMock.EXPECT().Shutdown(gomock.Any()).Return(nil, nil),
+			)
+
+			// When
+			err := sut.Shutdown(context.Background())
+
+			// Then
+			Expect(err).To(BeNil())
+		})
+	})
+
+	t.Describe("StopStreamServer", func() {
+		// Prepare the sut
+		BeforeEach(setupSUT)
+
+		It("should succeed", func() {
+			// Given
+			/* gomock.InOrder(
```

Comment:
Why is the commented out? Can we remove it if it is not needed?

Reply:
Yes, I will remove it.